### ISSUE TITLE
Services no longer supported

### ISSUE THREAD

**Issue by jdeath** [jdeath/Hubspace-Homeassistant#94]
Creating an issue to add back services. Worked before 3.0 release, but less necessary now. It was handy option to support less often used calls. 

Service allowed manually sending a functionClass, functionInstance, value to an entity

**Comment by Expl0dingBanana** [comment]
I can get this re-implemented sometime next week. Are there any features missing that should be available from HA UI?

**Comment by almoney** [comment]
My automations have not been working since the latest update.  I removed the configuration.yaml entried and then added Hubspace in integrations.  Manual switching works, but call service light off or on do not function and the light off/on automation also will not change the device state in Hubspace.  I can control manually.

**Comment by jdeath** [comment]
@almoney I think your issue is different. If you use native services of light/fan, etc it should work. I tested my light and it works. The only thing that does not work is sending a raw hubspace command.

Did your device names change? Use the Developer Tools->Service tab to get the right call. If those are fans, they are now fan.primary1fan and you need to use the fan.turn_on service. I do not have fans, so cannot test.

@Expl0dingBanana I do not really use any services, but it was a feature people used to set light effects, fan settings, etc. You may have integrated some of those into effects, but be nice to have option to manually send an unique command not everyone uses. I do not have any devices that require them.

**Comment by almoney** [comment]
> @almoney I think your issue is different. If you use native services of light/fan, etc it should work. I tested my light and it works. The only thing that does not work is sending a raw hubspace command.
> 
> Did your device names change? Use the Developer Tools->Service tab to get the right call. If those are fans, they are now fan.primary1fan and you need to use the fan.turn_on service. I do not have fans, so cannot test.
> 
> @Expl0dingBanana I do not really use any services, but it was a feature people used to set light effects, fan settings, etc. You may have integrated some of those into effects, but be nice to have option to manually send an unique command not everyone uses. I do not have any devices that require them.

I switched to device control in my automation. This works the service calls did not work.  These are just plain switches controlling fans.

**Comment by almoney** [comment]
I am also seeing the switches report off in HA when they are on in Hubspace after you have turned them on from HA.

**Comment by jdeath** [comment]
Provide a device dump so ExplodingBanana can take a look. That is a different issue.

I always have issues writing services calls manually. Using the developers UI to select the service, then choose the entity, test if works, they copy the YAML is my method. I tested with entity as the target and does work.

**Comment by almoney** [comment]
> Provide a device dump so ExplodingBanana can take a look. That is a different issue.
> 
> I always have issues writing services calls manually. Using the developers UI to select the service, then choose the entity, test if works, they copy the YAML is my method. I tested with entity as the target and does work.

3.0.4 has repaired the status of my switches.  They no longer "display off" when they are on.  Thank you!

**Comment by Expl0dingBanana** [comment]
@jdeath I will have to look at the difference between a service and action

https://www.home-assistant.io/blog/2024/08/07/release-20248/#goodbye-service-calls-hello-actions-

**Comment by jdeath** [comment]
I think they just renamed it, but not 100% sure...

**Comment by Expl0dingBanana** [comment]
I am hoping to get to this issue sometime this week (most likely tomorrow)

**Comment by Expl0dingBanana** [comment]
@jdeath I am going to be approaching this in two parts. The first part will be standardizing the classes (inheriting from a base entity class that all entities implement). Once that is done, I believe I can implement a generic service call within the base class. The standardization has begun in the branch `standardize-entities`.

**Comment by Expl0dingBanana** [comment]
Can yall double check to see if services still dont work? I was able to toggle my light on and off with this

```yaml

entity_id:
  - light.office_fan_light
action: light.turn_on
data: {}
target:
  entity_id: light.office_fan_light
```

**Comment by Expl0dingBanana** [comment]
I believe all required services are supported as of 4.1.0. Please re-open the issue if the problem persists!
